fuel-vm/src/interpreter/initialization.rs

```diff
@@ -70,15 +71,32 @@ where
 impl<S, Tx> Interpreter<S, Tx>
 where
     Tx: ExecutableTransaction,
-    <Tx as IntoChecked>::Metadata: CheckedMetadata,
+{
+    /// Initialize the VM for a predicate context
+    pub fn init_predicate_estimation(&mut self, transaction: Tx, balances: InitialBalances) -> bool {
+        self.context = Context::PredicateEstimation {
+            program: Default::default(),
+        };
+
+        let mut transaction = transaction;
+        transaction.prepare_init_predicate();
```

Comment:
We need to zero the `Input.gas_used` field for estimation but not for verification, so you need to use another method here